internal/security/sensitive_string.go

```diff
@@ -0,0 +1,54 @@
+package security
+
+import (
+	"encoding/json"
+)
+
+const redactionPlaceholder = "***"
+
+// SensitiveString is a type for handling sensitive information securely.
+// This helps to achieve the following goals:
+//  1. Prevent accidental logging of sensitive information.
+//  2. Provide controlled visibility (e.g., redacted output for String() or MarshalJSON()).
+//  3. Enable controlled access to the sensitive value when needed.
+type SensitiveString struct {
+	value string
+}
+
+// NewSensitiveString creates a new SensitiveString
+func NewSensitiveString(value string) *SensitiveString {
```

Comment:
```suggestion
func NewSensitiveString(value string) SensitiveString {
```
IMO, should be a value, same as normal string.

Reply:
Yeah, already fixed as well, will push the change.